test/Tester/StorageFacet/Feature.Implementations/TableStorageFeature.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans;
+using Orleans.Runtime;
+using Tester.StorageFacet.Abstractions;
+using Tester.StorageFacet.Infrastructure;
+
+namespace Tester.StorageFacet.Implementations
+{
+    public class TableStorageFeature<TState> : IStorageFeature<TState>, ILifecycleParticipant<GrainLifecyleStage>, IConfigurableStorageFeature
+    {
+        private IStorageFeatureConfig config;
+        private bool activateCalled;
+
+        public string Name => this.config.StateName;
+        public TState State { get; set; }
+
+        public Task Save()
+        {
+            return Task.CompletedTask;
+        }
+
+        public string GetExtendedInfo()
+        {
+            return $"Table:{this.Name}-ActivateCalled:{this.activateCalled}, StateType:{typeof(TState).Name}";
+        }
+
+        public Task LoadState(CancellationToken ct)
+        {
+            this.activateCalled = true;
+            return Task.CompletedTask;
+        }
+
+        public void Participate(ILifecycleObservable<GrainLifecyleStage> lifecycle)
+        {
+            lifecycle.Subscribe(GrainLifecyleStage.SetupState, LoadState);
+        }
+
+        public void Configure(IStorageFeatureConfig cfg)
+        {
+            this.config = cfg;
+        }
+    }
+
+    public class TableStorageFeatureFactory : StorageFeatureFactory
+    {
+        public TableStorageFeatureFactory(IGrainActivationContext context) : base(typeof(TableStorageFeature<>), context)
```

Comment:
I understand this is just test code, but still, I would expect the specific factory implementation be responsible of newing up (using ActivationServices or with the `new` keyword), instead of just passing the type up to the base class and only allowing the base class to use the container to resolve the type. This is already a factory, so it makes sense that it is the responsibility of this derived class to decide how it instantiates, instead of having the `StorageFeatureFactory.Create<TState>` implementation.

Also, my preference would be that this derived class instead of just using the container to resolve, that it uses the `new` keyword or the `ActivatorUtilities` class, but pass the configuration to the constructor, instead of requiring a separate step to pass the configuration to the feature.  With that, you can remove the `IConfigurableStorageFeature` interface.
It will also allow different implementations of the feature to participate in different stages of the activation lifecycle if they so require.